ui/server/src/test/scala/pl/touk/nussknacker/ui/validation/AdditionalPropertiesValidatorTest.scala

```diff
@@ -49,39 +66,101 @@ class AdditionalPropertiesValidatorTest extends FunSuite with Matchers {
         "propReq" -> ""
       ))
     ))
+
     val result = validator.validate(process)
 
-    checkOne(result, Seq("empty", "propReq"), Seq("type", "propReq"))
+    result.errors.processPropertiesErrors shouldBe List(
+      NodeValidationError(
+        "InvalidLiteralIntValue",
+        s"Property $reqFieldName ($label) has value of invalid type",
+        s"Expected integer, got: ''.",
+        fieldName = Some(reqFieldName),
+        errorType = NodeValidationErrorType.SaveNotAllowed),
+      NodeValidationError(
+        "EmptyMandatoryParameter",
+        s"Empty expression for mandatory parameter",
+        s"Please fill expression for this parameter",
+        Some(reqFieldName),
+        NodeValidationErrorType.SaveAllowed),
+    )
   }
 
-  test("validate non empty config with required property of invalid type") {
+  test("validate non empty config with required property with wrong type") {
     val process = ProcessTestData.displayableWithAdditionalFields(Some(
       ProcessAdditionalFields(None, Set.empty, properties = Map(
-        "propOpt" -> "a"
+        "propReq" -> "some text"
       ))
     ))
+
     val result = validator.validate(process)
 
-    checkOne(result, Seq("empty", "propReq"))
+    result.errors.processPropertiesErrors shouldBe List(
+      NodeValidationError(
+        "InvalidLiteralIntValue",
+        s"Property $reqFieldName ($label) has value of invalid type",
```

Comment:
Can you validate only type of error (without message). Let's make this tests less "concrete" and take advantage of error types.

Reply:
Done